nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/list/AbstractListProcessor.java

```diff
@@ -375,7 +375,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                         // If our determined timestamp is the same as that of our last listing, skip this execution as there are no updates
                         if (minTimestampToListMillis.equals(this.lastListedLatestEntryTimestampMillis)) {
                             context.yield();
-                            return;
+                            break;
```

Comment:
sorry, I just realize that now, with this fix we are performing an extra listing which is not needed. Maybe it is better to just have return as before, but prepend a `justElectedPrimaryNode = false;`. So I am thinking of substituting this line with:

```
justElectedPrimaryNode = false;
return;
```

what do you think @ijokarumawak @joewitt  @viazovskyi ?
Thanks